lib/common.sh

```diff
@@ -424,9 +424,9 @@ differs(){
 # Inputs:
 # - Full name of a Docker/podman/crictl image including tag
 #
-function container_image_pull_if_missing() {
+function pull_container_image_if_missing() {
   local IMAGE="$1"
-  if ! sudo "${CONTAINER_RUNTIME}" | grep -q "$IMAGE"; then
+  if [[ -z $(sudo "${CONTAINER_RUNTIME}" image ls "$IMAGE" | tail +2) ]]; then
```

Comment:
```suggestion
  if [[ -z $(sudo "${CONTAINER_RUNTIME}" image ls "$IMAGE" -q) ]]; then
```

Reply:
`-q` lists the image IDs:

```
$ sudo docker image ls -q                                                                                       
6d49f8ffbce9                                                                                                                                                                                                                                                        
182c83f40238                                                                                                                                                                                                                                                        
ff66cc3c4d73                                                                                                                            
...             
```

It will be hard to match against these.

The purpose of the `tail +2` command is to skip the header:

```
$ sudo docker image ls kindest/node
REPOSITORY     TAG       IMAGE ID       CREATED        SIZE
kindest/node   v1.22.0   8bc140d2ba34   6 weeks ago    933MB
kindest/node   v1.21.2   bd88cf3babd7   2 months ago   928MB
```
```
$ sudo docker image ls kindest/node | tail +2
kindest/node   v1.22.0   8bc140d2ba34   6 weeks ago    933MB
kindest/node   v1.21.2   bd88cf3babd7   2 months ago   928MB
```

The version of `podman` in our environment has `-n` option which skips the header, but our version of `docker` doesn't have this, so I have to use the tail command.

Reply:
This could work! I think it is less obvious that the purpose of the `-q` is to skip the header though. Maybe the `| tail +2` is clearer?

Reply:
`podman` has `-n` which skips the header, but our version of Docker doesn't have this sadly.

Reply:
oh, I though `-q` would work for all container run time. Please do disregard my comment.

Reply:
`-q` will work for all the runtimes - the purpose is to show image IDs and it happens to skip the header, so it would work, I just wonder if it's clearer with the tail command that we want to skip the header.

Reply:
I get it `-q` does not add any value in this case. In fact, hides some useful details.